### PR TITLE
Add NCNN_LOGE

### PR DESCRIPTION
**Pull request by maxint**
Use macros to print debug info as there is no console IO support in some enviroment, e.g. bare metal system like QTEE.

### CODE REVIEW

**Comment by nihui** [comment]
do not change exp->expf etc in src/layer/lstm.cpp

Add the NCNN_LOGE macro into platform.h.in. It is so simple, a dedicated header file is not necessary

File: src/gpu.cpp [hunk 1]
```diff
@@ -822,15 +822,15 @@ int create_gpu_instance()
...
         }
 
-        fprintf(stderr, "[%u %s]  queueC=%u[%u]  queueG=%u[%u]  queueT=%u[%u]\n", i, physicalDeviceProperties.deviceName,
+        NCNN_LOGE("[%u %s]  queueC=%u[%u]  queueG=%u[%u]  queueT=%u[%u]", i, physicalDeviceProperties.deviceName,
                 gpu_info.compute_queue_family_index, gpu_info.compute_queue_count,
                 gpu_info.graphics_queue_family_index, gpu_info.graphics_queue_count,
                 gpu_info.transfer_queue_family_index, gpu_info.transfer_queue_count);
 
-        fprintf(stderr, "[%u %s]  buglssc=%d  bugihfa=%d\n", i, physicalDeviceProperties.deviceName,
+        NCNN_LOGE("[%u %s]  buglssc=%d  bugihfa=%d\n", i, physicalDeviceProperties.deviceName,
```
Comment (nihui) [inline src/gpu.cpp:830]: extra newline here

File: src/gpu.cpp [hunk 2]
```diff
@@ -822,15 +822,15 @@ int create_gpu_instance()
...
                 gpu_info.bug_local_size_spec_const, gpu_info.bug_implicit_fp16_arithmetic);
 
-        fprintf(stderr, "[%u %s]  fp16p=%d  fp16s=%d  fp16a=%d  int8s=%d  int8a=%d\n", i, physicalDeviceProperties.deviceName,
+        NCNN_LOGE("[%u %s]  fp16p=%d  fp16s=%d  fp16a=%d  int8s=%d  int8a=%d\n", i, physicalDeviceProperties.deviceName,
```
Comment (nihui) [inline src/gpu.cpp:833]: extra newline here

**Comment by maxint** [comment]
> 
> 
> do not change exp->expf etc in src/layer/lstm.cpp
> 
> Add the NCNN_LOGE macro into platform.h.in. It is so simple, a dedicated header file is not necessary

Got it.

**Comment by maxint** [comment]
> 
> 
> do not change exp->expf etc in src/layer/lstm.cpp
> 
> Add the NCNN_LOGE macro into platform.h.in. It is so simple, a dedicated header file is not necessary

There is warning about implicit conversion from double to float when use exp() instead of expf().

**Comment by nihui** [comment]
> > do not change exp->expf etc in src/layer/lstm.cpp
> > Add the NCNN_LOGE macro into platform.h.in. It is so simple, a dedicated header file is not necessary
> 
> There is warning about implicit conversion from double to float when use exp() instead of expf().

static_cast<float>exp() or (float)exp()

**Comment by maxint** [comment]
https://github.com/Tencent/ncnn/pull/1753/checks?check_run_id=666522060
Why there is no build log.

**Comment by maxint** [comment]
https://github.com/Tencent/ncnn/pull/1753/checks?check_run_id=666522060
https://github.com/Tencent/ncnn/pull/1753/checks?check_run_id=666635829
The ios_cpu check should be passed.

**Comment by nihui** [comment]
Thanks for your contribution !

**Comment by maxint** [comment]
🤝

**Pull request merged**